utils/clients/bitflyer/client.go

```diff
@@ -241,9 +265,77 @@ func (c *HTTPClient) FetchQuote(
 	}
 	var body Quote
 	resp, err := c.client.Do(ctx, req, &body)
+	if err == nil {
+		expiry, err := parseExpiry(body.PriceToken)
+		if err == nil {
+			writeQuoteToFile(SavedQuote{
+				Body:   body,
+				Expiry: *expiry,
+			})
+		}
+	}
 	return &body, handleBitflyerError(err, req, resp)
 }
 
+// PriceTokenInfo holds info from the price token
+type PriceTokenInfo struct {
+	ProductCode string          `json:"product_code,omitempty"`
+	Rate        decimal.Decimal `json:"rate,omitempty"`
+	IssuedAt    int             `json:"iat,omitempty"`
+	Expiry      int             `json:"exp,omitempty"`
+}
+
+func parseExpiry(token string) (*time.Time, error) {
+	var claims map[string]interface{}
+	parsed, err := jwt.ParseSigned(token)
+	if err != nil {
+		return nil, err
+	}
+	err = parsed.UnsafeClaimsWithoutVerification(&claims)
```

Comment:
When we get the jwt key we should check the signature of the jwt.  Maybe we can make a ticket for future commit.

Reply:
https://github.com/brave-intl/bat-go/issues/706